common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -268,7 +309,261 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
+                wpa = 0x0;
+                if (bss_it->encryption_type != WSC::eWscEncr::WSC_ENCR_NONE) {
+                    LOG(ERROR) << "Autoconfiguration: " << vap << " encryption set on open VAP";
+                    abort = true;
+                    return;
+                }
+                if (bss_it->network_key.length() > 0) {
+                    LOG(ERROR) << "Autoconfiguration: " << vap << " network key set for open VAP";
+                    abort = true;
+                    return;
+                }
+            } else if (bss_it->authentication_type == WSC::eWscAuth::WSC_AUTH_WPA2PSK) {
+                wpa = 0x2;
+                wpa_key_mgmt.assign("WPA-PSK");
+                // Cipher must include AES for WPA2, TKIP is optional
+                if ((uint16_t(bss_it->encryption_type) & uint16_t(WSC::eWscEncr::WSC_ENCR_AES)) ==
+                    0) {
+                    LOG(ERROR) << "Autoconfiguration:  " << vap
+                               << " CCMP(AES) is required for WPA2";
+                    abort = true;
+                    return;
+                }
+                if ((uint16_t(bss_it->encryption_type) & uint16_t(WSC::eWscEncr::WSC_ENCR_TKIP)) !=
+                    0) {
+                    wpa_pairwise.assign("TKIP CCMP");
+                } else {
+                    wpa_pairwise.assign("CCMP");
+                }
+                if (bss_it->network_key.length() < 8 || bss_it->network_key.length() > 64) {
+                    LOG(ERROR) << "Autoconfiguration: " << vap << " invalid network key length "
+                               << bss_it->network_key.length();
+                    abort = true;
+                    return;
+                }
+                if (bss_it->network_key.length() < 64) {
+                    wpa_passphrase.assign(bss_it->network_key);
+                } else {
+                    wpa_psk.assign(bss_it->network_key);
+                }
+                ieee80211w.assign("0");
+                disable_pmksa_caching.assign("1");
+                okc.assign("0");
+                wpa_disable_eapol_key_retries.assign("0");
+            } else {
+                LOG(ERROR) << "Autoconfiguration: " << vap << " invalid authentication type";
+                abort = true;
+                return;
+            }
+
+            LOG(DEBUG) << "Autoconfiguration for ssid: " << bss_it->ssid
+                       << " auth_type: " << auth_type << " encr_type: " << enc_type
+                       << " network_key: " << bss_it->network_key
+                       << " fronthaul=" << beerocks::string_utils::bool_str(bss_it->fronthaul)
+                       << " backhaul=" << beerocks::string_utils::bool_str(bss_it->backhaul);
+
+            conf.set_create_vap_value(vap, "wps_state", bss_it->fronthaul ? "2" : "");
+            conf.set_create_vap_value(vap, "wps_independent", "0");
+            conf.set_create_vap_value(vap, "sFourAddrMode", bss_it->backhaul ? "1" : "");
```

Comment:
This one is Intel-specific, remove it.

I'm actually surprised this worked, since hostapd errors out on unkown config file options. Maybe you never tested with a bBSS?

Reply:
The latter. I did some very limited tests. We should defiantly do more tests, more environments and more scenarios.
I am lacking the knowledge of doing it correctly.